tests/flags_to_string_ut/flags_to_string_ut.c

```diff
@@ -0,0 +1,358 @@
+    va_end(va);
+    return result;
+}
+
+MU_DEFINE_ENUM_STRINGS(UMOCK_C_ERROR_CODE, UMOCK_C_ERROR_CODE_VALUES)
+static void on_umock_c_error(UMOCK_C_ERROR_CODE error_code)
+{
+    ASSERT_FAIL("umock_c reported error :%" PRI_MU_ENUM "", MU_ENUM_VALUE(UMOCK_C_ERROR_CODE, error_code));
+}
+
+
+static unsigned char va_list_to_string[] = "some va_list stringification";
+static char* umockvalue_stringify_va_list(const void* value)
+{
+    (void)value;
+    char* result = malloc(sizeof(va_list_to_string));
+    ASSERT_IS_NOT_NULL(result);
+    memcpy(result, va_list_to_string, sizeof(va_list_to_string));
+    return result;
+}
+
+static int umockvalue_are_equal_va_list(const void* left, const void* right)
+{
+    (void)memcmp(left, right, sizeof(va_list));
+    return 0;
+}
+
+static int umockvalue_copy_va_list(void* destination, const void* source)
+{
+    (void)memcpy(destination, source, sizeof(va_list));
+    return 0;
+}
+
+static void umockvalue_free_va_list(void* value)
+{
+    (void)value;
+    //free(value);
+}
+
+
+
+
+BEGIN_TEST_SUITE(TEST_SUITE_NAME_FROM_CMAKE)
+
+TEST_SUITE_INITIALIZE(suite_init)
+{
+    ASSERT_ARE_EQUAL(int, 0, real_gballoc_hl_init(NULL, NULL));
+
+    ASSERT_IS_NOT_NULL(test_serialize_mutex = TEST_MUTEX_CREATE());
+
+    ASSERT_ARE_EQUAL(int, 0, umock_c_init(on_umock_c_error), "umock_c_init failed");
+    ASSERT_ARE_EQUAL(int, 0, umocktypes_stdint_register_types(), "umocktypes_stdint_register_types failed");
+    ASSERT_ARE_EQUAL(int, 0, umocktypes_charptr_register_types(), "umocktypes_charptr_register_types failed");
+
+    REGISTER_GBALLOC_HL_GLOBAL_MOCK_HOOK();
+    REGISTER_STRING_UTILS_GLOBAL_MOCK_HOOK();
+
+    //linux says these days that sizeof(va_list) is 24, so REGISTER_UMOCK_ALIAS_TYPE(va_list, void*); would not work.
+
+    REGISTER_UMOCK_VALUE_TYPE(va_list, umockvalue_stringify_va_list, umockvalue_are_equal_va_list, umockvalue_copy_va_list, umockvalue_free_va_list);
+
+}
+
+TEST_SUITE_CLEANUP(suite_cleanup)
+{
+    umock_c_deinit();
+
+    TEST_MUTEX_DESTROY(test_serialize_mutex);
+
+    real_gballoc_hl_deinit();
+}
+
+TEST_FUNCTION_INITIALIZE(method_init)
+{
+    if (TEST_MUTEX_ACQUIRE(test_serialize_mutex))
+    {
+        ASSERT_FAIL("Could not acquire test serialization mutex.");
+    }
+
+    umock_c_negative_tests_init();
+    umock_c_reset_all_calls();
+}
+
+TEST_FUNCTION_CLEANUP(method_cleanup)
+{
+    umock_c_negative_tests_deinit();
+    TEST_MUTEX_RELEASE(test_serialize_mutex);
+}
+
+/*Tests_SRS_FLAGS_TO_STRING_02_007: [ If following the reset of the known flags, there are no bits set in argument, then FLAGS_TO_STRING(X) shall prepare a string using the format "%s%s...%s", where each pair of %s%s corresponds to a pair of separator_N/value_N; ]*/
+/*Tests_SRS_FLAGS_TO_STRING_02_008: [ FLAGS_TO_STRING(X) shall succeed and return a non-NULL string. ]*/
+TEST_FUNCTION(FLAGS_TO_STRING_with_0_flags_succeeds)
+{
+    ///arrange
+    char* result;
+
+    const char* format;
+
```

Comment:
nit: extra line here and everywhere after format

Reply:
nit ignored.